lib/paperclip/storage/filesystem.rb

```diff
@@ -35,7 +35,21 @@ def exists?(style_name = default_style)
 
       def flush_writes #:nodoc:
         @queued_for_write.each do |style_name, file|
+          unless @options[:override_path_permissions] == false
```

Comment:
This is a lot to be putting into this method. Can you extract these two sections into methods?